src/Symfony/Component/Notifier/Bridge/Lox24/Webhook/LOX24RequestParser.php

```diff
@@ -33,28 +45,165 @@ protected function getRequestMatcher(): RequestMatcherInterface
     /**
      * @throws RejectWebhookException
      */
-    protected function doParse(Request $request, #[\SensitiveParameter] string $secret): ?SmsEvent
+    protected function doParse(Request $request, #[\SensitiveParameter] string $secret): SmsEvent|RemoteEvent|null
     {
         $payload = $request->request->all() ?? [];
         $name = $payload['name'] ?? null;
         $data = $payload['data'] ?? [];
 
-        if ('sms.delivery' !== $name) {
-            throw new RejectWebhookException(400, 'Notification name is not \'sms.delivery\'');
+        if (!\in_array($name, self::SUPPORTED_EVENTS, true)) {
+            throw new RejectWebhookException(400, \sprintf('Unsupported LOX24 event type: "%s".', $name));
+        }
+
+        $missingFields = [];
+        if (!isset($payload['id'])) {
+            $missingFields[] = '"id"';
+        }
+        if (!isset($payload['name'])) {
+            $missingFields[] = '"name"';
+        }
+        if (!isset($payload['data']) || !\is_array($payload['data'])) {
+            $missingFields[] = '"data"';
+        }
+
+        if (!empty($missingFields)) {
+            throw new RejectWebhookException(406, 'Payload is malformed - missing required fields: '.implode(', ', $missingFields).'.');
         }
 
-        if (!isset($data['id'], $data['status_code'])) {
-            throw new RejectWebhookException(406, 'Payload is malformed.');
+        return $this->createEventFromPayload($name, $data, $payload);
+    }
+
+    private function createEventFromPayload(string $eventName, array $data, array $payload): SmsEvent|RemoteEvent|null
+    {
+        switch ($eventName) {
```

Comment:
could be more readable with a match
```suggestion
        return match ($eventName) {
```